samples/snippets/sample_default_values.py

```diff
@@ -0,0 +1,114 @@
+        report_name_prefix: Report Name Prefix which defaults to an empty string
+            to showcase default values behaviour.
+    """
+    usage_export_location = compute_v1.UsageExportLocation({
+        'bucket_name': bucket_name,
+        'report_name_prefix': report_name_prefix
+    })
+
+    if not report_name_prefix:
+        # Sending empty value for report_name_prefix, will result with the
+        # next usage report generated will have the default prefix value
+        # "usage_gce". (ref: https://cloud.google.com/compute/docs/reference/rest/v1/projects/get)
+        print("Setting report_name_prefix to empty value will cause the report "
+              "to have the default prefix of `usage_gce`.")
+
+    projects_client = compute_v1.ProjectsClient()
+    projects_client.set_usage_export_bucket(
+        project=project_id, usage_export_location_resource=usage_export_location)
+    return
+# [END compute_usage_report_set]
+
+
+# [START compute_usage_report_get]
+def get_usage_export_bucket(project_id: str) -> compute_v1.UsageExportLocation:
+    """
+    Retrieve Compute Engine usage export bucket for the Cloud Project.
+    Replaces the empty value returned by the API with the default value used
+    to generate report file names.
+
+    Params:
+        project_id: project ID or project number of the project to update.
+    Returns:
+        UsageExportLocation object describing the current usage export settings
+        for project project_id.
+    """
+    projects_client = compute_v1.ProjectsClient()
+    project_data = projects_client.get(project=project_id)
+
+    uel = project_data.usage_export_location
+
+    if not uel.bucket_name:
+        # The Usage Reports are disabled.
+        return uel
+    if not uel.report_name_prefix:
+        # Although the server sent the empty value, the next usage report
+        # generated with these settings still has the default prefix value
+        # "usage_gce". (ref: https://cloud.google.com/compute/docs/reference/rest/v1/projects/get)
+        print('Report name prefix not set, replacing with default value of '
+              '`usage_gce`.')
+        uel.report_name_prefix = 'usage_gce'
+    return uel
+# [END compute_usage_report_get]
+# [END compute_instances_verify_default_value]
+
+
+# [START compute_usage_report_disable]
+def disable_usage_export(project_id: str) -> None:
+    """
+    Disable Compute Engine usage export bucket for the Cloud Project.
+
+    Params:
+        project_id: project ID or project number of the project to update.
+    """
+    projects_client = compute_v1.ProjectsClient()
+    export_location = compute_v1.UsageExportLocation(
+        {'bucket_name': '', 'report_name_prefix': ''})
+
+    # Updating the setting with empty bucket name will disable the
+    # usage report generation.
+    projects_client.set_usage_export_bucket(
+        project=project_id, usage_export_location_resource=export_location)
```

Comment:
Please double check if this actually works. I've noticed that for PHP you have to send 'null' as usage_export_location_resource. Setting empty bucket_name has no effect on the GCP setting.

Reply:
Hey, you're right! I was assuming that [the docs](https://cloud.google.com/compute/docs/logging/usage-export#disabling_usage_reports) are correct - turns out they aren't! Thanks for this catch.